src/core/shape_inference/include/strided_slice_shape_inference.hpp

```diff
@@ -0,0 +1,222 @@
+
+        if (input_shape.rank().is_dynamic()) {
+            output_shapes[0] = ov::PartialShape::dynamic();
+            return;
+        }
+
+        auto input_rank = input_shape.size();
+
+        AxisSet new_axis_mask = convert_mask_to_axis_set(op->get_new_axis_mask());
+
+        NODE_VALIDATION_CHECK(op,
+                              input_rank + new_axis_mask.size() >= begin.size(),
+                              "Input rank plus number of new axis has to be at least the size of Lower "
+                              "and Upper bounds vector.");
+
+        AxisSet begin_mask = convert_mask_to_axis_set(op->get_begin_mask());
+        AxisSet end_mask = convert_mask_to_axis_set(op->get_end_mask());
+        AxisSet shrink_axis_mask = convert_mask_to_axis_set(op->get_shrink_axis_mask());
+
+        std::vector<DimType> dim;
+
+        int64_t input_shape_idx = 0;
+        for (size_t axis = 0; axis < begin.size(); ++axis) {
+            // add all dimensions hidden under the ellipsis mask if ellipsis mask is set
+            if (ellipsis_mask.count(axis)) {
+                // only one bit in ellipsis mask is allowed
+                int num_new_axis_after_ellipses = 0;
+                int num_input_axis_before_ellipses = 0;
+                for (size_t i = 0; i < axis; ++i) {
+                    if (!new_axis_mask.count(i)) {
+                        num_input_axis_before_ellipses++;
+                    }
+                }
+                for (size_t i = axis + 1; i < begin.size(); ++i) {
+                    if (new_axis_mask.count(i)) {
+                        num_new_axis_after_ellipses++;
+                    }
+                }
+
+                int64_t num_input_axis_after_ellipses =
+                    (begin.size() - axis - num_new_axis_after_ellipses - 1);  // -1 because it's a position of ellipses
+                int64_t num_of_hidden_dims =
+                    input_rank - num_input_axis_after_ellipses - num_input_axis_before_ellipses;
+                for (int64_t i = 0; i < num_of_hidden_dims; ++i) {
+                    dim.emplace_back(input_shape[input_shape_idx]);
+                    input_shape_idx++;
+                }
+            } else {
+                // add new single dimension if new_axis_mask is set
+                if (new_axis_mask.count(axis)) {
+                    dim.emplace_back(1);
+                }
+                // skip this dimension if shrink_axis_mask is set
+                else if (shrink_axis_mask.count(axis)) {
+                    input_shape_idx++;
+                }
+                // calculating dimension (begin, end, begin_mask, end_mask, stride)
+                else {
+                    // check dynamic dimension
+                    if (input_shape[input_shape_idx].is_dynamic()) {
+                        input_shape_idx++;
+                        dim.emplace_back(Dimension::dynamic());
+                        continue;
+                    }
```

Comment:
Looks like we can handle some cases with dynamic dimensions: e.g if we have dimension as an `interval(0, K)` and `end[i] = N`, where `N < K` we will have output dimension with `interval(0,N)` (when `strides[i] == 1`)

Reply:
Added!